include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,204 @@
+                T(-1.0), v, w2, T(1.0), std::move(c));
+}
+
+// Implementation based on:
+// 1. Algorithm 6 "LAPACK Algorithm for the eigenvector back-transformation", page 15, PhD thesis
+// "GPU Accelerated Implementations of a Generalized Eigenvalue Solver for Hermitian Matrices with
+// Systematic Energy and Time to Solution Analysis" presented by Raffaele Solcà (2016)
+// 2. G. H. Golub and C. F. Van Loan, Matrix Computations, chapter 5, The Johns Hopkins University Press
+template <class T>
+struct BackTransformation<Backend::MC, Device::CPU, T> {
+  static void call_FC(Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  if (m <= 1 || n == 0)
+    return;
+
+  const SizeType nr_reflector = mat_v.size().rows() - mb;
+
+  dlaf::matrix::Distribution dist_t({mb, nr_reflector}, {mb, mb});
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsV(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsT(n_workspaces, dist_t);
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;
+  const SizeType last_tile_cols = mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols();
+
+  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
+    bool is_last = (k == num_panel_refls - 1) ? true : false;
+    const SizeType t_last_cols = dist_t.tileSize(GlobalTileIndex{k, k}).cols();
+
+    auto& panelV = panelsV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+    auto& panelT = panelsT.nextResource();
+
+    panelV.setRangeStart({k + 1, k});
+    panelW.setRangeStart({k + 1, k});
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      // Copy V panel into VV

```

Comment:
Moreover, about comments, I think this is potentially misleading.

You are copying just the head of the panel because it is not well formed (i.e. it should be lower triangular), but for the rest you are just linking the panel to use the tiles in-place (because it is enough a readonly access).